src/main/java/io/ebeaninternal/server/deploy/BeanDescriptorManager.java

```diff
@@ -567,7 +567,7 @@ private void initialiseAll() {
 
   private void checkForValidEmbeddedId(BeanDescriptor<?> d) {
     IdBinder idBinder = d.getIdBinder();
-    if (idBinder != null && idBinder instanceof IdBinderEmbedded) {
+    if (idBinder instanceof IdBinderEmbedded) { // If variable is null, will result in false
```

Comment:
Actually adding in the comment // If variable ...   means I can't review this once.  You should leave out all comments like that from the code and put them in the PR instead if you like.